docs/data/data-grid/aggregation/aggregation.md

```diff
@@ -4,15 +4,277 @@ title: Data Grid - Aggregation
+This means that the data from groups explicitly provided in your dataset will be ignored in favour of their aggregated values.  
+:::
+
+The demo below shows the _sum_ aggregation on the "Size" column and the _max_ aggregation on the "Last modification" column.
+
+{{"demo": "AggregationTreeData.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Filtering
+
+By default, the aggregation only uses the filtered rows.
+You can set the `aggregatedRows` to `"all"` to use all rows.
+
+In the example below, the movie _Avatar_ is not passing the filters but is still used for the **max** aggregation of the `gross` column.
+
+{{"demo": "AggregationFiltering.js", "bg": "inline"}}
+
+## Aggregation functions
+
+### Basic structure
+
+An aggregation function is an object describing how to generate an aggregated value for a given set of values.
+
+Let's take a look at a simple aggregation function:
+
+The full typing details can be found on the [GridAggregationFunction API page](/x/api/data-grid/grid-aggregation-function/).
+
+### Built-in functions
+
+The `@mui/x-data-grid-premium` package comes with a set of built-in aggregation functions to cover the basic use-cases:
+
+| Name   | Behavior                                                   | Supported column types       |
+| ------ | ---------------------------------------------------------- | ---------------------------- |
+| `sum`  | Returns the sum of all values in the group                 | `number`                     |
+| `avg`  | Returns the non-rounded average of all values in the group | `number`                     |
+| `min`  | Returns the smallest value of the group                    | `number`, `date`, `dateTime` |
+| `max`  | Returns the largest value of the group                     | `number`, `date`, `dateTime` |
+| `size` | Returns the amount of cells in the group                   | all                          |
+
+### Remove a built-in function
+
+#### Remove a built-in function for all columns
+
+You can remove some aggregation functions for all columns by passing a filtered object to the `aggregationFunctions` prop.
+
+In the example below, the `sum` aggregation function has been removed.
+
+{{"demo": "AggregationRemoveFunctionAllColumns.js", "bg": "inline"}}
+
+#### Remove a built-in function for one column
+
+You can remove some aggregation function for one column by passing a `availableAggregationFunctions` property to the column definition.
+
+```ts
+const column = {
+  field: 'year',
+  type: 'number',
+  availableAggregationFunctions: ['max', 'min'],
+};
+```
+
+In the example below, the only aggregation function available for the **Year** column is `max` whereas all aggregation functions are available for the **Gross** column
+
+{{"demo": "AggregationRemoveFunctionOneColumn.js", "bg": "inline"}}
+
+### Create a custom functions
+
+You can pass custom aggregation functions to the `aggregationFunctions` prop.
+
+An aggregation function is an object with the following shape:
+
```

Comment:
I think we can try displaying a dummy object before showing the concrete example.
Also, maybe a concrete example using numbers would be more practical. 

I understand the value of bringing a "different" function though. Perhaps we can have both since we have the second example already.

Reply:
I added a simplified version on the `min` aggregation function as a 1st very basic example